lib/tasks/wikirate/importer.rb

```diff
@@ -0,0 +1,107 @@
+class Importer
+  attr_reader :export_location
+  def initialize location
+    @export_location =
+      case location
+      when "dev"    then "dev.wikirate.org"
+      when "demo"   then "demo.wikirate.org"
+      when "local"  then "localhost:3000"
+      else               "wikirate.org"
+      end
+  end
+
+  # @return [Array<Hash>] each hash contains the attributes for a card
+  def items_of cardname, opts={}
+    card_data =
+      work_on "getting data from #{cardname} card" do
+        if opts[:subitems]
+          json_export cardname, :export_items
+        else
+          json_export(cardname)["card"]["value"]
+        end
+      end
+    import_card_data card_data
+  end
+
+  def cards_of_type type
+    items_of "#{type}+*type+by_update"
+  end
+
+  def migration_records
+    migration_data =
+      work_on "getting migration records" do
+        json_export :admin_info, :migrations
+      end
+    work_on "importing migration records" do
+      import_migration_data migration_data
+    end
+  end
+
+  private
+
+  def work_on msg
+    puts msg.green
+    result = yield
+    puts " ... done".green
+    result
+  end
+
+  def json_export cardname, view=nil
+    name = cardname.is_a?(Symbol) ? ":#{cardname}" : cardname.to_name.key
+    url = "http://#{@export_location}/#{name}.json"
+    url += "?view=#{view}" if view
+    JSON.parse open(url, read_timeout: 50_000).read
+  end
+
+  def update_or_create name, _codename, attr
+    if attr["type"].in? %w(Image File)
+      attr["content"] = ""
+      attr["empty_ok"] = true
+    end
+    begin
+      # card = codename ? Card.fetch(codename.to_sym) : Card.fetch(name)
+      card = Card.fetch(name)
+      if card
+        puts "updating card #{name} "\
+               "#{card.update_attributes!(attr)}".light_blue
+      else
+        puts "creating card #{name} #{Card.create!(attr)}".yellow
+      end
+    rescue => e
+      puts "Error in #{name}\n#{e}".red
+    end
+  end
```

Comment:
no urgent comments here, but long term:  since there's type-specific stuff, I was wondering whether we might want there to be a card method (eg #import) that we could override.

eg
```
def import attrib={}
  assign_attributes attrib
  save!
end
```